lib/ohai/mixin/ec2_metadata.rb

```diff
@@ -202,6 +202,20 @@ def metadata_key(key)
         key.gsub(/\-|\//, "_")
       end
 
+      # @param [String] data that might be JSON
+      #
+      # @return [Boolean] is the data JSON or not?
+      def json?(data)
+        data = StringIO.new(data)
```

Comment:
Instead of creating the objects twice why not just change the above if / elseif to rescue to the 2nd condition? It would save a lot of duplicate object setup. Also this same issue exists in gce mixin as well from my quick glance.

Reply:
There's got to be some sort of jinx when you review the same thing without seeing it because github didn't update while you were typing out a comment.